binding.gyp

```diff
@@ -78,7 +79,8 @@
           # supports ALPN. The target is "[major].[minor].[patch]". We split by
           # periods and take the first field to get the major version.
         'defines': [
-          'TSI_OPENSSL_ALPN_SUPPORT=<!(echo <(target) | cut -d. -f1)'
+          'TSI_OPENSSL_ALPN_SUPPORT=<!(echo <(target) | cut -d. -f1)',
+          'GPR_BACKWARDS_COMPATIBILITY_MODE'
```

Comment:
Masood: everything is good currently, and will be after this PR for Linux
Python. We don't want to use the binary compatibility hacks with other c
core builds, and those by caller are going to be more than the wrapped
languages.

So... The default for c core being binary backwards compatible needs to
change (which it is here), and each wrapped language needs to be tweaked to
set the now non default option.

On Thu, Feb 4, 2016, 4:21 PM Craig Tiller <ctiller@google.com> wrote:

> I get that :)
>
> Wasn't sure what the syntax was for 'always'
>
> On Thu, Feb 4, 2016, 3:35 PM Michael Lumish <notifications@github.com>
> wrote:
>
>> In binding.gyp
>> <https://github.com/grpc/grpc/pull/5078#discussion_r51956219>:
>>
>> > @@ -78,7 +79,8 @@
>> >            # supports ALPN. The target is "[major].[minor].[patch]". We split by
>> >            # periods and take the first field to get the major version.
>> >          'defines': [
>> > -          'TSI_OPENSSL_ALPN_SUPPORT=<!(echo <(target) | cut -d. -f1)'
>> > +          'TSI_OPENSSL_ALPN_SUPPORT=<!(echo <(target) | cut -d. -f1)',
>> > +          'GPR_BACKWARDS_COMPATIBILITY_MODE'
>>
>> I just want to point out that you're doing this in both the "if" case and
>> "else" case of a conditional expression.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/grpc/grpc/pull/5078/files#r51956219>.
>>
>